### PR TITLE
Create offsite payments app extension template

### PR DESCRIPTION
**Pull request by marklevi**
### Background

Related issue: https://github.com/Shopify/payments-platform/issues/4378
Related project: https://docs.google.com/document/d/1Y2Ly5sw6MalZQP8w3yraQlOs8dK3UK38LVQvQz7fC_M/edit#heading=h.8coln95uazox

Creates an offsite payments app extension template to be used via the CLI

### Checklist

- [ ] I have :tophat:'d these changes
- [x] I have squashed my commits into chunks of work with meaningful commit messages

### CODE REVIEW

**Review by wikiti — APPROVED** [review]

File: payments-app-extension-offsite/shopify.extension.toml.liquid [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+start_refund_session_url = "https://api.paymentprovider.com/refund"
+start_capture_session_url = "https://api.paymentprovider.com/capture"
+start_void_session_url = "https://api.paymentprovider.com/void"
+supported_countries = ["US"]
+supported_payment_methods = ["visa"]
+supports_3ds = false
+supports_installments = false
+supports_deferred_payments = false
+test_mode_available = true
+api_version = "2021-07"
```
[the remark this answers is not in the document]
Comment (marklevi) [reply payments-app-extension-offsite/shopify.extension.toml.liquid:22]: what makes you think it is deprecated? even though we use a default value, shouldnt we leave the option for the partner to set this field?

File: payments-app-extension-offsite/shopify.extension.toml.liquid [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+multiple_capture = false
```
[the remark this answers is not in the document]
Comment (marklevi) [reply payments-app-extension-offsite/shopify.extension.toml.liquid:23]: I dont see why it's necessary but I can add a newline

**Comment by shopiwags** [comment]
👋 If this is just truly for offsite - we don't use any extensibility stuff on that payment method so the template is likely ok here.
 
However, if you plan to re use this for custom onsite at all then yeah we should have the proper extensibility props in here as well.

File: payments-app-extension-offsite/shopify.extension.toml.liquid [hunk 3]
```diff
@@ -0,0 +1,23 @@
+[[extensions]]
+name = "{{ name }}"
+type = "payments_app_extension"
+handle = "{{ handle }}"
+target = "offsite"
```
Comment (isaacroldan) [inline payments-app-extension-offsite/shopify.extension.toml.liquid:5]: Also, other extensions define the target inside a `targetting` object like:

```toml
  [[extensions.targeting]]
  target = "...."
```

Did you discuss this structure with anyone from app foundations?

Comment (marklevi) [reply payments-app-extension-offsite/shopify.extension.toml.liquid:11]: I posted a request to review this PR on their slack channel: https://shopify.slack.com/archives/C01D7U7AJ5C/p1704886259783989

**Pull request merged**